docs/log-correlation.asciidoc

```diff
@@ -0,0 +1,96 @@
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/ruby[elastic.co]
+endif::[]
+
+[[log-correlation]]
+== Log correlation
+
+Trace/log correlation can be set up in three different ways.
```

Comment:
Perhaps a link to the three different options? They're all on this page but it might improve scanability

```suggestion
Trace/log correlation can be set up in three different ways:

* <<rails-tagged-logging>>
* <<lograge>>
* <<manually-formatting-logs>>
```